core/src/main/scala/com/salesforce/op/filters/RawFeatureFilter.scala

```diff
@@ -146,31 +150,36 @@ class RawFeatureFilter[T]
       val predOut = allPredictors.map(TransientFeature(_))
       (respOut, predOut)
     }
-    val preparedFeatures: RDD[PreparedFeatures] = data.rdd.map(PreparedFeatures(_, responses, predictors, timePeriod))
-
-    implicit val sgTuple2Maps = new Tuple2Semigroup[Map[FeatureKey, Summary], Map[FeatureKey, Summary]]()
+    val preparedFeatures: RDD[PreparedFeatures] =
+      data.rdd.map(PreparedFeatures(_, responses, predictors, timePeriod))
+    val allFeatures: RDD[AllFeatures] = preparedFeatures.map(_.allFeatures)
+    val textFeatures: RDD[Map[FeatureKey, Seq[String]]] = allFeatures.map(_._3)
+    val (totalCount, responseSummaries, numericSummaries, textSummaries) =
+      RawFeatureFilter.getAllSummaries(bins, allFeatures)
     // Have to use the training summaries do process scoring for comparison
-    val (responseSummaries, predictorSummaries): (Map[FeatureKey, Summary], Map[FeatureKey, Summary]) =
-      allFeatureInfo.map { info =>
-        info.responseSummaries -> info.predictorSummaries
-      }.getOrElse(preparedFeatures.map(_.summaries).reduce(_ + _))
-    val (responseSummariesArr, predictorSummariesArr): (Array[(FeatureKey, Summary)], Array[(FeatureKey, Summary)]) =
-      (responseSummaries.toArray, predictorSummaries.toArray)
-
-    implicit val sgTuple2Feats = new Tuple2Semigroup[Array[FeatureDistribution], Array[FeatureDistribution]]()
-    val (responseDistributions, predictorDistributions): (Array[FeatureDistribution], Array[FeatureDistribution]) =
-      preparedFeatures
-        .map(_.getFeatureDistributions(
-          responseSummaries = responseSummariesArr,
-          predictorSummaries = predictorSummariesArr,
-          bins = bins,
-          textBinsFormula = textBinsFormula,
-          featureDistributionType
-        )).reduce(_ + _)
+    val responseDistributions: Map[FeatureKey, FeatureDistribution] =
+      responseSummaries.map { case (key, sum) =>
+        key -> sum.getFeatureDistribution(key, totalCount, featureDistributionType)
+      }
+    val numericDistributions: Map[FeatureKey, FeatureDistribution] =
+      numericSummaries.map { case (key, sum) =>
+        key -> sum.getFeatureDistribution(key, totalCount, featureDistributionType)
+      }
+
+    // This will initialize existing text summary hashing TFs
+    textSummaries.foreach { case (_, textSum) => textSum.setHashingTF() }
```

Comment:
Perhaps call setHashingTF when creating a new TextSummary (i.e., in updateTextSummaries)?